main.js

```diff
@@ -0,0 +1,161 @@
+// router
+const links = document.querySelector(".header__bot-links").querySelectorAll("a")
+
+function changeActiveLink() {
+  for (const link of links) {
+    if (location.hash === "") {
+      links[0].classList.add("header__bot-link--active")
+      break
+    }
+    if (link.href === location.href)
+      link.classList.add("header__bot-link--active")
+    else link.classList.remove("header__bot-link--active")
+  }
+}
+
+const route = (event) => {
+  event.preventDefault()
+  history.pushState(null, null, event.target.href)
+  handleLocation()
+}
+
+for (const link of links) {
+  link.addEventListener("click", route)
+}
+
+const routes = {
+  "": "./pages/activity.html",
+  activity: "./pages/activity.html",
+  map: "./pages/map.html",
+  timer: "./pages/timer.html",
+  404: "./pages/404.html",
+}
+
+const handleLocation = async () => {
+  const path = location.hash.slice(1)
+  const route = routes[path] || routes[404]
+  const html = await fetch(route).then((data) => data.text())
+  const root = document.querySelector("#root")
+  root.innerHTML = html
+  changeActiveLink()
+
+  if (path === "timer") timer()
+  else clearInterval(pageTimerId)
+
```

Comment:
тут не хватает "защиты от пользователя", ты можешь не один раз перейти на одинаковую страницу, например с таймером, и он у тебя будет каждый раз создаваться, но очищаться будет один раз при выходе
```suggestion
  if (pageTimerId) {
    clearInterval(pageTimerId)
    pageTimerId = null
  }
  
  if (path === "timer") timer()
```